tests/version/test_markers.py

```diff
@@ -171,6 +171,22 @@ def test_single_marker_intersect() -> None:
     assert str(intersection) == 'python_version >= "3.4" and python_version < "3.6"'
 
 
+@pytest.mark.parametrize(
+    ("marker1", "marker2"),
+    [
+        ('sys_platform == "win32"', 'sys_platform == "linux"'),
+        ('sys_platform != "win32"', 'sys_platform == "win32"'),
+        ('python_version >= "3.6"', 'python_version < "3.6"'),
+        ('python_version == "3.6"', 'python_version == "3.7"'),
+        ('python_version > "3.6"', 'python_version <= "3.6"'),
+        ('python_version >= "3.6"', 'python_version < "3.6"'),
+        ('python_version > "3.6"', 'python_version < "3.7"'),
```

Comment:
The last test is the relevant one but since I did not find such a basic test function that tests for empty single marker intersections, I just added some additional tests.